lib/runtime/unknown_transaction.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// UnknownTransaction is child VDT of TransactionValidityError
+type UnknownTransaction scale.VaryingDataType
+
+// Index returns the VDT index
+func (UnknownTransaction) Index() uint {
+	return 1
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (u *UnknownTransaction) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*u)
+	err = vdt.Set(val)
+	if err != nil {
+		return err
+	}
+	*u = UnknownTransaction(vdt)
+	return nil
+}
+
+// Value will return value from underying VaryingDataType
+func (u *UnknownTransaction) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*u)
+	return vdt.Value()
+}
+
+func (u UnknownTransaction) Error() string {
+	value := u.Value()
+	if value == nil {
+		return "unknownTransaction hasn't been set"
+	}
+	err, ok := value.(error)
+	if !ok {
+		panic(fmt.Sprintf("%T does not implement the error type", value))
+	}
+	return err.Error()
+}
+
+// NewUnknownTransaction is constructor for Unknown
+func NewUnknownTransaction() UnknownTransaction {
+	vdt, err := scale.NewVaryingDataType(ValidityCannotLookup{}, NoUnsignedValidator{}, UnknownCustom(0))
+	if err != nil {
+		panic(err)
+	}
+	return UnknownTransaction(vdt)
+}
+
+// ValidityCannotLookup Could not look up some information that is required to validate the transaction
+type ValidityCannotLookup struct{}
```

Comment:
I think this could possibly be nice, maybe after investigating all the cases some of these are used in substrate (and if possible to consolidate). I think it is outside of the scope of this PR though

Reply:
Do you mind creating an issue for this? Thanks!